bool printCapabilities
bool printOutputs
bool printControls
---
